### PR TITLE
sys/ztimer: cleanup xtimer_compat.h for 32Bit only

### PR DESCRIPTION
**Pull request by kfessel**

### CODE REVIEW

File: sys/include/ztimer/xtimer_compat.h [hunk 1]
```diff
@@ -158,14 +145,24 @@ static inline uint32_t xtimer_usec_from_ticks(xtimer_ticks32_t ticks)
...
+{
+    return usec;
+}
+
 static inline void xtimer_now_timex(timex_t *out)
 {
-    uint64_t now = xtimer_now_usec64();
+    uint32_t now = xtimer_now_usec();
 
     out->seconds = div_u64_by_1000000(now);
```
Comment (fjmolinas) [inline sys/include/ztimer/xtimer_compat.h:167]: Expects 64 bit, I don't think the proposed change fits the api expectation.

File: sys/include/ztimer/xtimer_compat.h [hunk 2]
```diff
@@ -205,12 +202,7 @@ static inline void xtimer_set_timeout_flag64(xtimer_t *t, uint64_t timeout)
 
 static inline void xtimer_spin(xtimer_ticks32_t ticks)
 {
-    assert(ticks < US_PER_MS);
-    ztimer_now_t start = ztimer_now(ZTIMER_USEC);
-
-    while (ztimer_now(ZTIMER_USEC) - start < ticks) {
-        /* busy waiting */
-    }
+    ztimer_spin(ZTIMER_USEC,xtimer_usec_from_ticks(ticks));
```
Comment (fjmolinas) [inline sys/include/ztimer/xtimer_compat.h:205]:
```suggestion
    ztimer_spin(ZTIMER_USEC, xtimer_usec_from_ticks(ticks));
```

**Review by fjmolinas — COMMENTED** [review]
I feel this can be its own PR

**Comment by kfessel** [comment]
moved this to https://github.com/RIOT-OS/RIOT/pull/17690

**Comment by kfessel** [comment]
moved this to https://github.com/RIOT-OS/RIOT/pull/17690

**Pull request closed without merging**